app/controllers/submit.js

```diff
@@ -333,8 +338,38 @@ export default Ember.Controller.extend(BasicsValidations, NodeActionsMixin, Tagg
             this.send('next', this.get('_names.0'));
         },
         resetFileUpload() {
-            this.send('changeState', State.START);
-            this.set('fileAndNodeLocked', false);
+            var promisesArray = [];
+            var filePromises = [];
+            this.get('filesUploadedForPreprint').forEach((file) => {
+                filePromises.push(this.get('fileManager').deleteFile(file).then(() => {
+                    this.get('toast').info('File removed!');
+                }).catch(() => {
+                    this.get('toast').error('Could not remove uploaded file');
+                }));
+            });
+
+            Ember.RSVP.all(filePromises).then(() => {
+                this.get('projectsCreatedForPreprint').forEach((project) => {
+                    promisesArray.push(this.store.adapterFor('node').ajax(project.get('links.self'), 'DELETE').then(() => {
+                        this.get('toast').info('Project removed!');
+                    }).catch(() => {
+                        this.get('toast').error('Could not remove created project/component');
+                    }));
+                });
+            });
+
+            Ember.RSVP.all(promisesArray).then(() => {
+                if (promisesArray.length > 0 || filePromises.length > 0) {
+                    window.setTimeout(
+                    function() {
+                        location.reload();
+                    }, 3000);
```

Comment:
The purpose of the 3 second wait was just so the user could see the success messages.  This branch isn't hit until the promises have resolved.